.github/workflows/code-cover-gen.yml

```diff
@@ -0,0 +1,77 @@
+name: PR code coverage (generate)
+on:
+  pull_request:
+    types: [ opened, reopened, synchronize ]
+    branches: [ master ]
+
+jobs:
+  code-cover-gen:
+    runs-on: ubuntu-latest
+    env:
+      PR: ${{ github.event.pull_request.number }}
+      HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # By default, checkout merges the PR into the current master.
+          # Instead, we want to check out the PR as is.
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 10
+
+      - name: Set up Bazel cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/bazel
+          key: ${{ runner.os }}-bazel-${{ hashFiles('.bazelversion', '.bazelrc', 'WORKSPACE', 'WORKSPACE.bazel', 'MODULE.bazel') }}
+          restore-keys: |
+            ${{ runner.os }}-bazel-    
+
+      - name: Get list of changed packages
+        shell: bash
+        run: |
+          set -euxo pipefail
+          # To get the base commit, we get the number of commits in the PR.
+          # Note that github.event.pull_request.base.sha is not what we want,
+          # that is the tip of master and not necessarily the PR fork point.
+          NUM_COMMITS=$(gh pr view $PR --json commits --jq '.commits | length')
+          # This limit must not exceed the checkout fetch-depth above.
+          if [ ${NUM_COMMITS} -ge 10 ]; then
+            echo "PR #$PR has too many commits (${NUM_COMMITS})"
+            exit 1
+          fi
+          BASE_SHA=$(git rev-parse HEAD~${NUM_COMMITS})
+          CHANGED_PKGS=$(build/ghactions/changed-go-pkgs.sh ${BASE_SHA} ${HEAD_SHA})
+          if [[ $(echo "${CHANGED_PKGS}" | wc -w) -gt 20 ]]; then
+            echo "PR #$PR changes too many packages; skipping code coverage"
+            exit 1
```

Comment:
If we're going to do this, then you should `exit 0` here so merging your PR is not blocked. Sometimes you have to change 21+ packages at once.